### ISSUE TITLE
Unable to set empty JSON object

### ISSUE THREAD

**Issue by Pascalio** [lmorg/murex#507]
**Describe the bug:**
Cannot store an object whose content has been filtered away.
```
set o = %{thing:{stuff:chow}}
$o -> pretty -> ![ thing ] -> set o 
#Error in `set` (0,30): cannot store variable: unexpected end of JSON input

```
This is however necessary when you have a list of things to be done and you want to remove things as they get completed; it might (hopefully) happen that the list goes empty, but we still have to store that.

**Comment by lmorg** [comment]
Looks like the bug lies with `pretty`.

The following stuff works as expected:
```
» o = %{thing:{stuff:chow}}
» $o -> ![ thing ]

» set json o = %{thing:{stuff:chow}}
» $o -> ![ thing ]
```

The following doesn't work because it is missing a type annotation:
```
» set o = %{thing:{stuff:chow}}
» $o -> ![ thing ]
Error in `![` (0,7): Parameter, `thing`, isn't an integer. strconv.Atoi: parsing "thing": invalid syntax
```
(the error message is a little unclear here but essentially because the data type is a string and multiline strings are automatically treated as lists, it is expecting an integer to filter).

`pretty` does set its output data type to `json` but there's something glitchy about `pretty` because it hangs the shell if piped to `debug`.

You shouldn't really need to use `pretty` for assignments within murex because the shell should natively switch pretty formatting when writing to a TTY. `pretty` is only intended for reformatting output from external commands. So can work around this problem by dropping `set` from your declaration (or annotating `set` if you want to be explicit) and thus removing the need to use `pretty`. That all said, `pretty` behaving the way it does is absolutely a bug and will still need to be rectified.

**Comment by Pascalio** [comment]
Thanks for your message!
Indeed, my use of `pretty` was the result of my forgetting how to assign stuff correctly... My bad, sorry! 
Actually, this bug report here was not about a type problem, but rather that I can't assign the filtered-to-void output. To put it in (updated) code:
```
o = %{thing:{stuff:chow}}
$o -> ![ thing ] -> set b
# Error in `set` (0,21): cannot store variable: unexpected end of JSON input
```
The problem isn't in `$o -> ![ thing ]` but in the `-> set b` part...

**Comment by lmorg** [comment]
My apologies, I understand now.

I will take a look at this on Sunday.

**Comment by Pascalio** [comment]
No worries, thank you so much!

**Comment by lmorg** [comment]
Code in `develop`, will be merged into v3.1
